File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/latest/contract/erp/BaseErpContractTest.java
package com.mycompany.myshop.systemtest.latest.contract.erp;

import com.mycompany.myshop.systemtest.latest.contract.base.BaseExternalSystemContractTest;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

public abstract class BaseErpContractTest extends BaseExternalSystemContractTest {
    @Test
    void shouldBeAbleToGetProduct() {
        assertDoesNotThrow(() -> scenario
                .given().product().withSku("BOOK-123").withUnitPrice(12.0)
                .then().product("BOOK-123").hasSku("BOOK-123").hasPrice(12.0));
    }
}

File: system-test/typescript/tests/latest/contract/erp/BaseErpContractTest.ts
import type { ContractTest } from '../base/BaseExternalSystemContractTest.js';

export function registerErpContractTests(test: ContractTest): void {
  test('shouldBeAbleToGetProduct', async ({ scenario }) => {
    await scenario
      .given()
      .product()
      .withSku('BOOK-123')
      .withUnitPrice(12)
      .then()
      .product('BOOK-123')
      .hasSku('BOOK-123')
      .hasPrice(12);
  });
}
